crates/polars-expr/src/expressions/cast.rs

```diff
@@ -13,7 +13,14 @@ pub struct CastExpr {
 
 impl CastExpr {
     fn finish(&self, input: &Series) -> PolarsResult<Series> {
-        input.cast_with_options(&self.dtype, self.options)
+        let dtype = if matches!(input.dtype(), DataType::List(_)) && !self.dtype.is_nested() {
```

Comment:
Question for reviewer: this seems like the wrong place to do it. Is there a better place?